File: client/src/app/modules/quotation/services/building.domiciliation.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { BuildingDomiciliation } from '../../quotation/model/BuildingDomiciliation';

@Injectable({
  providedIn: 'root'
})
export class BuildingDomiciliationService extends AppRestService<BuildingDomiciliation>{

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getBuildingDomiciliations() {
    return this.getListCached(new HttpParams(), "building-domiciliations");
  }

  addOrUpdateBuildingDomiciliation(buildingDomiciliation: BuildingDomiciliation) {
    this.clearListCache(new HttpParams(), "building-domiciliations");
    return this.addOrUpdate(new HttpParams(), "building-domiciliation", buildingDomiciliation, "Enregistré", "Erreur lors de l'enregistrement");
  }
}

File: src/main/java/com/jss/osiris/modules/osiris/quotation/service/BuildingDomiciliationService.java
package com.jss.osiris.modules.osiris.quotation.service;

import java.util.List;

import com.jss.osiris.modules.osiris.quotation.model.BuildingDomiciliation;

public interface BuildingDomiciliationService {
    public List<BuildingDomiciliation> getBuildingDomiciliations();

    public BuildingDomiciliation getBuildingDomiciliation(Integer id);

    public BuildingDomiciliation addOrUpdateBuildingDomiciliation(BuildingDomiciliation buildingDomiciliation);

}
